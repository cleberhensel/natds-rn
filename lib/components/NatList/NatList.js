import React, { useState } from 'react';
import {
  View,
  FlatList,
} from 'react-native';
import PropTypes from 'prop-types';

import {
  NatListItem,
} from '../NatListItem';

export const NatList = ({
  items,
  selected,
  onChange,
  multiSelect,
  showHighlight,
}) => {
  const [currentSelected, setCurrentState] = useState(selected);

  const onPress = (callback, item) => {
    let result = currentSelected;
    if (multiSelect) {
      if (currentSelected.includes(item.id)) {
        result = currentSelected.filter(select => select !== item.id);
      } else {
        result = [...currentSelected, ...[item.id]];
      }
    } else if (currentSelected.includes(item.id)) {
      result = currentSelected.filter(select => select !== item.id);
    } else {
      result = [...[item.id]];
    }
    setCurrentState(result);
    if (callback) {
      callback(result);
    }
    if (item.onPress) {
      item.onPress(item);
    }
  };

  const renderItem = item => (
    <NatListItem
      title={item.title}
      iconLeft={item.iconLeft}
      iconRight={item.iconRight}
      onPressLeft={item.onPressLeft}
      onPressRight={item.onPressRight}
      hideIconLeft={item.hideIconLeft}
      hideIconRight={item.hideIconRight}
      onPress={() => onPress(onChange, item)}
      dividerTop={item.dividerTop}
      dividerBottom={item.dividerBottom}
      active={(currentSelected.includes(item.id) && showHighlight)}
    />
  );

  return (
    <View>
      <FlatList
        data={items}
        renderItem={({ item }) => renderItem(item)}
        keyExtractor={item => item.id}
      />
    </View>
  );
};

NatList.propTypes = {
  selected: PropTypes.arrayOf(
    PropTypes.string,
  ),
  onChange: PropTypes.func,
  multiSelect: PropTypes.bool,
  showHighlight: PropTypes.bool,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string,
      node: PropTypes.node,
      iconLeft: PropTypes.shape(),
      hideIconLeft: PropTypes.bool,
      iconRight: PropTypes.shape(),
      hideIconRight: PropTypes.bool,
      onPressRight: PropTypes.func,
      onPressLeft: PropTypes.func,
      onPress: PropTypes.func,
      dividerTop: PropTypes.bool,
      dividerBottom: PropTypes.bool,
    }),
  ).isRequired,
};

NatList.defaultProps = {
  selected: [],
  onChange: () => {},
  multiSelect: false,
  showHighlight: true,
};
