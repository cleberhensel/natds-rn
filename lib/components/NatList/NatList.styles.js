import styled from 'styled-components/native';
import { ThemeNaturaLight } from '../..';

export const HeaderWrapper = styled.View`
  flex: 1;
  width: 100%;
  height: 48;
  padding: 0px;
  position: relative;
  background-color: ${ThemeNaturaLight.palette.background.paper};
  flex-direction: row;
  align-items: stretch;
`;

export const HeaderTitle = styled.Text`
  margin-left: 0px;
  font-size: 16px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 16px;
  color: ${ThemeNaturaLight.palette.text.primary};
`;

export const IconPress = styled.TouchableOpacity`
  align-self: center;
  flex: 1;
  height: 40;
  width: 40;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-right: ${({ position }) => ((position === 'right') ? '16px' : '0px')};
  padding-left: ${({ position }) => ((position === 'right') ? '0px' : '16px')};
`;

export const TouchableOpacity = styled.TouchableOpacity`
  flex: 1;
`;

export const InputIcon = styled.Image`
  height: 24;
  width: 24;
`;
