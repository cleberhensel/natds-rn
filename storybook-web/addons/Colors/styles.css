.colors {
  font-weight: 400;
  line-height: 1.5;
}

.colors__container {
  max-width: 1400px;
  display: flex;
  flex-flow: row wrap;
  padding: 60px;
}

.colors__group {
  display: flex;
  flex-flow: row wrap;
  flex: 0 0 100%;
  margin-bottom: 30px;
}

.colors__group__title {
  font-size: 1.5rem;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-weight: 400;
  color: #555;
  flex: 0 0 100%;
  padding: 0 0 20px;
  text-transform: capitalize;
  margin: 0;
}

.colors__group__list {
  flex: 0 0 100%;
  display: flex;
  flex-flow: row wrap;
  padding: 0;
  margin: 0;
}

.colors__item {
  width: 130px;
  display: flex;
  flex: 0 0 auto;
  position: relative;
  flex-flow: column;
  overflow: hidden;
  padding: 0;
  margin: 0 15px 15px 0;
}

.colors__item__color {
  width: 130px;
  height: 130px;
  margin-bottom: 5px;
}

.colors__item__color-name,
.colors__item__name {
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-size: 0.875rem;
  line-height: 1rem;
  padding: 0;
  margin: 0;
}

.colors__item__name {
  color: #888;
  margin-bottom: 5px;
}

.colors__item__color-name {
  color: #555;
}
