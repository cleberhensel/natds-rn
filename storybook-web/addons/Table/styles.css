.table__container {
  font-weight: 400;
  line-height: 1.5;
  max-width: 800px;
  width: 100%;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 60px;
}

.table__title {
  font-size: 18px;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 0 20px;
  margin: 0;
}

.table {
  width: 100%;
  border-collapse: collapse;
}

.table__body__row {
  background: #f9f9f9;
  border-bottom: #fff 1px solid;
  padding: 10px;
}

.table__head__item:first-child {
  width: 25%;
}

.table__head__item,
.table__body__item {
  font-size: 0.875rem;
  text-align: left;
}

.table__head__item {
  padding-bottom: 10px;
  text-transform: capitalize;
}

.table__body__item {
  line-height: 1.25rem;
  font-weight: 400;
  height: 20px;
  padding: 10px;
}

.table__body__item__color {
  border: #ddd 2px solid;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  display: inline-block;
  vertical-align: top;
  margin: 2px 10px 0;
}

.table__body__item__color-name {
  line-height: 1rem;
  display: inline-block;
}
