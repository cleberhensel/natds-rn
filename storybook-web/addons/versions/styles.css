.versions {
  font-weight: 400;
  line-height: 1.5;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 30px;
}

.versions__type {
  padding: 0 0 10px;
  margin: 0 10px 10px;
}

.versions__type__name {
  border-bottom: rgba(0,0,0,.1) 1px solid;
  font-size: 18px;
  text-transform: capitalize;
  padding: 0 0 10px;
  margin: 0 0 10px;
}

.versions__type__warning {
  font-size: 18px;
  text-transform: capitalize;
  margin-bottom: 10px;
}

.versions__type__button {
  background: rgba(0,0,0,.3);
  border: 0;
  border-radius: 4px;
  font-size: 1rem;
  color: #fff;
  text-transform: capitalize;
  cursor: pointer;
  padding: 7px 14px;
  margin: 0 10px 10px 0;
}

.versions__type__button:hover {
  background: rgba(0,0,0,.2);
}

.versions__type__button:active {
  background: rgba(0,0,0,.25);
}

.versions__type__button--selected,
.versions__type__button--selected:hover {
  background: rgba(0,0,0,.6);
  cursor: default;
}
