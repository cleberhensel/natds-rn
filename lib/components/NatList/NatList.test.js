import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { NatList } from './NatList';
import { HeaderTitle } from './NatList.styles';

describe('Header component', () => {
  const listItems = [
    {
      id: '1',
      title: 'Title Text',
      iconLeft: { svg: { name: 'outlined-navigation-menu' } },
      iconRight: { svg: { name: 'outlined-action-autofilter' } },
      onPressLeft: () => alert('press'),
      onPress: () => alert('press'),
      onPressRight: () => alert('press'),
    },
    {
      id: '2',
      title: 'Title Text',
      iconRight: { svg: { name: 'outlined-action-autofilter' } },
      iconLeft: false,
      onPressLeft: () => alert('press'),
      onPressRight: () => alert('press'),
      dividerBottom: true,
    },
    {
      id: '3',
      title: 'Title Text',
      iconLeft: { svg: { name: 'outlined-navigation-menu' } },
      iconRight: false,
      onPressLeft: () => alert('press'),
      onPressRight: () => alert('press'),
    },
    {
      id: '4',
      title: 'Title Text',
      iconLeft: { svg: { name: 'outlined-navigation-menu' } },
      iconRight: false,
      onPressLeft: () => alert('press'),
      onPressRight: () => alert('press'),
      dividerTop: true,
    },
  ];

  it('Should render container component correctly', () => {
    const header = renderer.create(
      <NatList
        items={listItems}
      />,
    ).toJSON();

    expect(header).toMatchSnapshot();
  });

  it('Should render header component with title', () => {
    shallow(
      <NatList
        items={listItems}
        select="1"
      />,
    );
  });
});
