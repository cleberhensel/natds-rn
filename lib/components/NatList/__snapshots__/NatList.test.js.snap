// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render container component correctly 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
>
  <View>
    <View
      accessibilityRole="header"
      accessible={true}
      active={false}
      dividerBottom={false}
      dividerTop={false}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF29",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 0,
              "paddingTop": 12,
              "width": 40,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
        </View>
      </View>
      <View
        hideIconLeft={false}
        hideIconRight={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": 0,
              "color": "#333333",
              "justifyContent": "center",
              "left": 48,
              "right": 48,
              "top": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="Title Text"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingTop": 12,
                },
              ]
            }
          >
            Title Text
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 0,
              "paddingRight": 16,
              "paddingTop": 12,
              "width": 40,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
        </View>
      </View>
    </View>
    <View
      accessibilityRole="header"
      accessible={true}
      active={false}
      dividerBottom={true}
      dividerTop={false}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF29",
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 0,
              "paddingTop": 12,
              "width": 40,
            }
          }
        />
      </View>
      <View
        hideIconLeft={false}
        hideIconRight={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": 0,
              "color": "#333333",
              "justifyContent": "center",
              "left": 48,
              "right": 48,
              "top": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="Title Text"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingTop": 12,
                },
              ]
            }
          >
            Title Text
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 0,
              "paddingRight": 16,
              "paddingTop": 12,
              "width": 40,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
        </View>
      </View>
    </View>
    <View
      accessibilityRole="header"
      accessible={true}
      active={false}
      dividerBottom={false}
      dividerTop={false}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF29",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 0,
              "paddingTop": 12,
              "width": 40,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
        </View>
      </View>
      <View
        hideIconLeft={false}
        hideIconRight={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": 0,
              "color": "#333333",
              "justifyContent": "center",
              "left": 48,
              "right": 48,
              "top": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="Title Text"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingTop": 12,
                },
              ]
            }
          >
            Title Text
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 0,
              "paddingRight": 16,
              "paddingTop": 12,
              "width": 40,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityRole="header"
      accessible={true}
      active={false}
      dividerBottom={false}
      dividerTop={true}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF29",
            "borderTopColor": "#EEEEEE",
            "borderTopWidth": 1,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 0,
              "paddingTop": 12,
              "width": 40,
            }
          }
        >
          <SvgMock
            height={24}
            style={
              Object {
                "fill": "#333333",
              }
            }
            width={24}
          />
        </View>
      </View>
      <View
        hideIconLeft={false}
        hideIconRight={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": 0,
              "color": "#333333",
              "justifyContent": "center",
              "left": 48,
              "right": 48,
              "top": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityLabel="Title Text"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingTop": 12,
                },
              ]
            }
          >
            Title Text
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 40,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 0,
              "paddingRight": 16,
              "paddingTop": 12,
              "width": 40,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
